docs/enhancements/improved-handling-for-pre-installation-errors.md

```diff
@@ -0,0 +1,128 @@
+---
+title: Improved handling for pre installation errors
+authors:
+  - "@pmaidmen"
+creation-date: 2023-07-29
+last-updated: 2023-07-29
+---
+
+# Summary
+
+When a cluster installation is triggered, either via Assisted Installer KubeAPI or via the UI, there are preparation steps that take place before the installation is launched. Presently, not all of these failures are appropriately reported back to users and there is a lack of consistency.
+The aim of this enhancement is to come up with a set of changes that improve this situation greatly.
+Please see MGMT-12397 for more background on this.
+
+# Issues identified
+
+There are a number of issues that contribute to a lack of clarity for the end user.
+
+* There is no guarantee that every preinstallation failure will be reported back to the user, either via KubeAPI or via the UI.
+* Some parts of the preinstallation depend on other conditions, at least with these the cause of the error can be somewhat determined.
+* The UI does not show enough detail in the error summary for the user to have a clue why an error occurred.
+* Inconsistent error handling here https://github.com/openshift/assisted-service/blob/master/internal/bminventory/inventory.go#L1293-L1360
+    - Sure, every error is handled but not enough is done to ensure this is comminicated in a useful way to the user
+    - Also inconsistent terminology is used here, making it unclear if it's important that we are failing a Preparation or a Preinstallation
+* If using a KubeAPI based implementation such as MCE then the installation can get into an "ininite loop" because the statemachine transitions back to "ready" when a Pre Installation error occurs. We should ideally constrain this some so that fruitless attempts can be halted.
+    
+
+# Motivation
+
+We want to improve the general user experience in this area so that customers will quickly be able to fix common problems with less assistance.
+
+### Goals
+
+- In KubeAPI - Constrain installation to a maximum number of attenmpts then hold the installation to prevent fruitless retries.
+- In KubeAPI - When an installation is held after a maximum number of attempts - Set a condition to explain the reason of the last failure.
+- In the Cluster Object - add an additional field to store the reason for a preinstallation failure.
+- In cluster validation, add a validation to show the status of cluster preinstallation and any errors.
+- Ensure that the UI is updated appropriately to display failure information.
+
+### Non-Goals
+
+- TBC
+
+### Implementation proposal 
+
+    - Add a cluster "pre installation status" validation
```

Comment:
How does the failing validation succeed again so that the user can retry?

Reply:
Did we get any input from UX on this?  How would this kind of failure be displayed ideally?

@nirfarkas